src/linear_algebra/matrix/is_diag.lean

```diff
@@ -0,0 +1,197 @@
+/-
+Copyright (c) 2021 Lu-Ming Zhang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Lu-Ming Zhang
+-/
+import linear_algebra.matrix.symmetric
+import linear_algebra.matrix.orthogonal
+import data.matrix.kronecker
+
+/-!
+# Diagonal matrices
+
+This file contains the definition and basic results about diagonal matrices.
+
+## Main results
+
+- `matrix.is_diag`: a proposition that stats a given square matrix `A` is diagonal
+                    (i.e. `∀ m n, i ≠ j → A m n = 0`).
+
+## Tags
+
+diag, diagonal, matrix
+-/
+
+namespace matrix
+
+variables {α β R n m : Type*}
+
+open function
+open_locale matrix kronecker
+
+/-- `A.is_diag` means square matrix `A` is a dianogal matrix: `∀ i j, i ≠ j → A i j = 0`. -/
```

Comment:
```suggestion
/-- `A.is_diag` means square matrix `A` is a diagoal matrix. -/
```
No need to include the definition, there are other ways to see that. (also this fixes a typo)